NextcloudTalk/NCSettingsController.m

```diff
@@ -555,24 +481,34 @@ - (BOOL)isGuestsAppEnabled
 - (void)subscribeForPushNotificationsForAccountId:(NSString *)accountId
 {
 #if !TARGET_IPHONE_SIMULATOR
-    NCPushNotificationKeyPair *keyPair = [self generatePushNotificationsKeyPairForAccountId:accountId];
-
+    NCPushNotificationKeyPair *keyPair = nil;
+    NSData *pushNotificationPublicKey = [[NCKeyChainController sharedInstance] pushNotificationPublicKeyForAccountId:accountId];
+    NSData *pushNotificationPrivateKey = [[NCKeyChainController sharedInstance] pushNotificationPrivateKeyForAccountId:accountId];
+    
+    if (pushNotificationPublicKey && pushNotificationPrivateKey) {
+        keyPair = [[NCPushNotificationKeyPair alloc] init];
+        keyPair.publicKey = pushNotificationPublicKey;
+        keyPair.privateKey = pushNotificationPrivateKey;
+    } else {
+        keyPair = [self generatePushNotificationsKeyPairForAccountId:accountId];
+    }
+    
     if (!keyPair) {
-        NSLog(@"Error while subscribing: Unable to generate push notifications key pair.");
+        [NCUtils log:@"Error while subscribing: Unable to generate push notifications key pair."];
```

Comment:
Differentiate between unable to generate and no keyPair available?

Reply:
Left it like that as discussed.